packages/subgraph-service/contracts/libraries/LegacyAllocation.sol

```diff
@@ -39,19 +39,22 @@ library LegacyAllocation {
     /**
      * @notice Migrate a legacy allocation
      * @dev Requirements:
-     * - The allocation must not exist
+     * - The allocation must not have been previously migrated
      * @param self The legacy allocation list mapping
      * @param indexer The indexer that owns the allocation
      * @param allocationId The allocation id
      * @param subgraphDeploymentId The subgraph deployment id the allocation is for
+     * @custom:error LegacyAllocationAlreadyMigrated if the allocation has already been migrated
      */
     function migrate(
         mapping(address => State) storage self,
         address indexer,
         address allocationId,
         bytes32 subgraphDeploymentId
     ) internal {
-        require(!self[allocationId].exists(), LegacyAllocationExists(allocationId));
+        if (self[allocationId].exists()) {
+            revert LegacyAllocationAlreadyMigrated(allocationId);
+        }
```

Comment:
This should be equivalent and matches the error throwing pattern we are using (use require and throw custom errors).

```suggestion
        require(!self[allocationId].exists(), LegacyAllocationAlreadyMigrated(allocationId);
```

Reply:
Okay, apparently require is more gas efficient than if revert? But happy to change this back

Reply:
To be honest I'm not sure which one is more gas efficient 😄 
Since we are on Arbitrum One where gas is no longer a huge problem, I'd rather have it be more readable regardless of the gas cost in this case.